presto-main/src/test/java/com/facebook/presto/operator/TestPartitionedOutputOperator.java

```diff
@@ -0,0 +1,340 @@
+        partitionedOutputOperator.addInput(pageWithRunLength);
+        OperatorContext operatorContext = partitionedOutputOperator.getOperatorContext();
+        long outputSize = operatorContext.getOutputDataSize().getTotalCount();
+        long outputPositionCount = operatorContext.getOutputPositions().getTotalCount();
+        assertEquals(outputSize, 13_599);
+        assertEquals(outputPositionCount, 1_511);
+        assertTrue(outputSize > pageWithRunLength.getSizeInBytes());
+    }
+
+    private PartitionedOutputOperator createPartitionedOutputOperator(boolean shouldReplicate)
+    {
+        PartitionFunction partitionFunction = new LocalPartitionGenerator(new InterpretedHashGenerator(ImmutableList.of(BIGINT), new int[] {0}), PARTITION_COUNT);
+        PagesSerdeFactory serdeFactory = new PagesSerdeFactory(new BlockEncodingManager(new TypeRegistry()), false);
+        OutputBuffers buffers = createInitialEmptyOutputBuffers(PARTITIONED);
+        for (int partition = 0; partition < PARTITION_COUNT; partition++) {
+            buffers = buffers.withBuffer(new OutputBuffers.OutputBufferId(partition), partition);
+        }
+        PartitionedOutputBuffer buffer = createPartitionedBuffer(
+                buffers.withNoMoreBufferIds(),
+                new DataSize(Long.MAX_VALUE, BYTE)); // don't let output buffer block
+        buffer.registerLifespanCompletionCallback(ignore -> {});
+        PartitionedOutputOperator.PartitionedOutputFactory operatorFactory = null;
+        if (shouldReplicate) {
+            operatorFactory = new PartitionedOutputOperator.PartitionedOutputFactory(
+                    partitionFunction,
+                    ImmutableList.of(0),
+                    ImmutableList.of(Optional.empty()),
+                    true,
+                    OptionalInt.of(0),
+                    buffer,
+                    new DataSize(1, GIGABYTE));
+        }
+        else {
+            operatorFactory = new PartitionedOutputOperator.PartitionedOutputFactory(
+                    partitionFunction,
+                    ImmutableList.of(0),
+                    ImmutableList.of(Optional.empty(), Optional.empty()),
+                    false,
+                    OptionalInt.empty(),
+                    buffer,
+                    new DataSize(1, GIGABYTE));
+        }
+
+        return (PartitionedOutputOperator) operatorFactory
+                .createOutputOperator(0, new PlanNodeId("plan-node-0"), TYPES, Function.identity(), serdeFactory)
+                .createOperator(createDriverContext());
+    }
+
+    private Page copyPositions(Page page, List<Integer> positions)
+    {
+        Block[] blocks = new Block[page.getChannelCount()];
+        for (int i = 0; i < blocks.length; i++) {
+            blocks[i] = page.getBlock(i).copyPositions(positions.stream().mapToInt(j -> j).toArray(), 0, positions.size());
+        }
+        return new Page(positions.size(), blocks);
+    }
+
+    private Page createSimplePage()
```

Comment:
There is only one line in this method. Let's inline it.

Reply:
This was for readability, since there is no Performance advantage with this change I hope this is ok to leave as is.